sdk/core/azure-core-http-vertx/src/main/java/com/azure/core/http/vertx/VertxAsyncHttpClientBuilder.java

```diff
@@ -0,0 +1,306 @@
+    }
+
+    /**
+     * Sets proxy configuration.
+     *
+     * @param proxyOptions The proxy configuration to use.
+     * @return The updated VertxAsyncHttpClientBuilder object.
+     */
+    public VertxAsyncHttpClientBuilder proxy(ProxyOptions proxyOptions) {
+        this.proxyOptions = proxyOptions;
+        return this;
+    }
+
+    /**
+     * Sets the configuration store that is used during construction of the HTTP client.
+     * <p>
+     * The default configuration store is a clone of the {@link Configuration#getGlobalConfiguration() global
+     * configuration store}, use {@link Configuration#NONE} to bypass using configuration settings during construction.
+     *
+     * @param configuration The configuration store.
+     * @return The updated VertxAsyncHttpClientBuilder object.
+     */
+    public VertxAsyncHttpClientBuilder configuration(Configuration configuration) {
+        this.configuration = configuration;
+        return this;
+    }
+
+    /**
+     * Sets custom {@link HttpClientOptions} for the constructed {@link io.vertx.core.http.HttpClient}.
+     *
+     * @param httpClientOptions The options of the web client.
+     * @return The updated VertxAsyncHttpClientBuilder object
+     */
+    public VertxAsyncHttpClientBuilder httpClientOptions(HttpClientOptions httpClientOptions) {
+        this.httpClientOptions = httpClientOptions;
+        return this;
+    }
+
+    /**
+     * Sets a custom {@link Vertx} instance that the constructed {@link io.vertx.core.http.HttpClient} will be created with.
+     *
+     * @param vertx The vertx instance.
+     * @return The updated VertxAsyncHttpClientBuilder object
+     */
+    public VertxAsyncHttpClientBuilder vertx(Vertx vertx) {
+        this.vertx = vertx;
+        return this;
+    }
+
+    /**
+     * Creates a new Vert.x {@link HttpClient} instance on every call, using the
+     * configuration set in the builder at the time of the build method call.
+     *
+     * @return A new Vert.x backed {@link HttpClient} instance.
+     */
+    public HttpClient build() {
+        boolean shutdownHookRequired = false;
+        if (this.vertx == null) {
+            ServiceLoader<VertxProvider> vertxProviders = ServiceLoader.load(VertxProvider.class, VertxProvider.class.getClassLoader());
+            Iterator<VertxProvider> iterator = vertxProviders.iterator();
+            if (iterator.hasNext()) {
+                VertxProvider provider = iterator.next();
+                this.vertx = provider.createVertx();
+                LOGGER.verbose("Using {} as the VertxProvider.", provider.getClass().getName());
+
+                while (iterator.hasNext()) {
+                    VertxProvider ignoredProvider = iterator.next();
+                    LOGGER.warning("Multiple VertxProviders were found on the classpath, ignoring {}.",
+                        ignoredProvider.getClass().getName());
+                }
+            } else {
+                this.vertx = Vertx.vertx();
```

Comment:
Even though this may be expensive I'm not sure we want to set the instance variable here or should use a local variable and leave `this.vertx` as null.